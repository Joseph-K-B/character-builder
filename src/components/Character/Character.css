.image-section{
    background-color: #5D536B;
}

.head-image{
    position: relative;
    top: 6.0%;
    left:34.0%;
    height: 5.5rem;
    width: 5.5rem;
    background-size: 5.5rem;
    background-repeat: no-repeat;
}

.torso-image{
    position: relative;
    top: 1.5%;
    left: 23.5%;
    height: 15.5rem;
    width: 15.5rem;
    background-size: 12.5rem;
    background-repeat: no-repeat;
}

.bottom-image{
    position: relative;
    top: -18.5%;
    left: 18.0%;
    height: 15.5rem;
    width: 15.5rem;
    background-size: 15.5rem;
    background-repeat: no-repeat;
}