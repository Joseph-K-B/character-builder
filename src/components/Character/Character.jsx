import React from "react";

function Character({
    head,
    setHead,
    torso,
    setTorso,
    bottom,
    setBottom
}) {
    return (
        <section>
            <figure
            aria-label='head'
            data-testid='head'
            value={head}
            onChange={(e) => setHead(e.target.value)}
            style={{
                height: '5.5rem',
                width: '5.5rem',
                backgroundSize: '5.5rem',
                backgroundImage: `url(./${head}-head.png)`,
                backgroundRepeat: 'no-repeat' 
              }}>
            </figure>
            <figure
            aria-label='torso'
            data-testid='torso'
            value={torso}
            onChange={(e) => setTorso(e.target.value)}
            style={{
                height: '15rem',
                width: '15rem',
                backgroundSize: '10rem',
                backgroundImage: `url(./${torso}-torso.png)`,
                backgroundRepeat: 'no-repeat' 
              }}>
            </figure>
            <figure
            aria-label='bottom'
            data-testid='bottom'
            value={bottom}
            onChange={(e) => setBottom(e.target.value)}
            style={{
                height: '15rem',
                width: '15rem',
                backgroundSize: '10rem',
                backgroundImage: `url(./${bottom}-bottom.png)`,
                backgroundRepeat: 'no-repeat' 
              }}>
            </figure>
        </section>
    )
}

export default Character;