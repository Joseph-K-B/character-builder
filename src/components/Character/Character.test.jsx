import { render, screen } from "@testing-library/react";
import Character from "./Character";

it('should render all elements on character page', () => {
    const pres = render(<Character />);
    const headImg = screen.getByTestId(/head/);
    const torsoImg = screen.getByTestId(/torso/);
    const bottomImg = screen.getByTestId(/bottom/);
    expect(headImg).toBeInTheDocument();
    expect(torsoImg).toBeInTheDocument();
    expect(bottomImg).toBeInTheDocument();
    expect (pres).toMatchSnapshot();
});
