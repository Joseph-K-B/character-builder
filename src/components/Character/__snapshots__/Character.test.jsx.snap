// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render all elements on character page 1`] = `
<div>
  <section>
    <figure
      aria-label="head"
      data-testid="head"
      style="height: 5.5rem; width: 5.5rem; background-size: 5.5rem; background-image: url(./undefined-head.png); background-repeat: no-repeat;"
    />
    <figure
      aria-label="torso"
      data-testid="torso"
      style="height: 15rem; width: 15rem; background-size: 10rem; background-image: url(./undefined-torso.png); background-repeat: no-repeat;"
    />
    <figure
      aria-label="bottom"
      data-testid="bottom"
      style="height: 15rem; width: 15rem; background-size: 10rem; background-image: url(./undefined-bottom.png); background-repeat: no-repeat;"
    />
  </section>
</div>
`;
